sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -80,13 +87,45 @@ case class CreateTableLikeCommand(
         s"Source table in CREATE TABLE LIKE cannot be temporary: '$sourceTable'")
     }
 
-    val tableToCreate = catalog.getTableMetadata(sourceTable).copy(
-      identifier = targetTable,
-      tableType = CatalogTableType.MANAGED,
-      createTime = System.currentTimeMillis,
-      lastAccessTime = -1).withNewStorage(locationUri = None)
+    val sourceTableDesc = catalog.getTableMetadata(sourceTable)
+
+    sourceTableDesc.tableType match {
+      case CatalogTableType.MANAGED | CatalogTableType.EXTERNAL | CatalogTableType.VIEW => // OK
+      case o => throw new AnalysisException(
+        s"CREATE TABLE LIKE is not allowed when the source table is ${o.name}")
+    }
 
-    catalog.createTable(tableToCreate, ifNotExists)
+    // For EXTERNAL_TABLE, the table properties has a particular field. To change it
+    // to a MANAGED_TABLE, we need to remove it; Otherwise, it will be EXTERNAL_TABLE,
+    // even if we set the tableType to MANAGED
+    // (metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java#L1095-L1105)
+    // Table comment is stored as a table property. To clean it, we also should remove them.
+    val newTableProp =
+      sourceTableDesc.properties.filterKeys(key =>
+        key != "EXTERNAL" && key.toLowerCase != "comment")
```

Comment:
I think we can remove these 2 tricks after https://github.com/apache/spark/pull/14155 and https://github.com/apache/spark/pull/14550